framework/permissions_test.py

```diff
@@ -21,35 +21,87 @@
 import mock
 import werkzeug.exceptions  # Flask HTTP stuff.
 
+from google.appengine.api import users
+
+from framework import basehandlers
 from framework import permissions
 
 
-class MockHandler(object):
+class MockHandler(basehandlers.BaseHandler):
 
   def __init__(self):
     self.called_with = None
 
   @permissions.require_admin_site
   def do_get(self, *args):
     self.called_with = args
+    return {'message': 'did get'}
 
+  @permissions.require_admin_site
+  def do_post(self, *args):
+    self.called_with = args
+    return {'message': 'did post'}
 
-class CanAdminSiteTests(unittest.TestCase):
 
-  def test_can_admin_site__normal_user(self):
-    """A normal user is not allowed to administer the site."""
-    testing_config.sign_in('user@example.com', 123)
-    self.assertFalse(permissions.can_admin_site())
+test_app = basehandlers.FlaskApplication(
+    [('/path', MockHandler),
+     ],
+    debug=True)
+
 
-  def test_can_admin_site__admin_user(self):
-    """An admin user is allowed to administer the site."""
+class PermissionFunctionTests(unittest.TestCase):
+
+  def check_function_results(
+      self, func, additional_args,
+      normal='missing', special='missing', admin='missing', anon='missing'):
+    """Test func under four conditions and check expected results."""
+    # Test normal users
+    testing_config.sign_in('user@example.com', 123)
+    user = users.get_current_user()
+    self.assertEqual(normal, func(user, *additional_args))
+
+    # Test special users
+    testing_config.sign_in('user@google.com', 123)
```

Comment:
A TODO here (and two lines down) to replace @google/@chromium references with scalable solution.